src/Vlingo.Common/Message/AsyncMessageQueue.cs

```diff
@@ -69,21 +69,14 @@ public virtual void Enqueue(IMessage message)
 
         public virtual void Flush()
         {
-            try
+            while (!queue.IsEmpty)
             {
-                while (!queue.IsEmpty)
-                {
-                    Thread.Sleep(1);
-                }
-
-                while (dispatching.Get())
-                {
-                    Thread.Sleep(1);
-                }
+                resetEvent.WaitOne(1);
```

Comment:
this `resetEvent` is shared everywhere in the class. The primary purpose was to communicate that:
- one task has been processed, get the next one 
- there is something in the queue, start processing now

now, with this change, here is what might happen. When the queue is `Flush`-ing and there are few messages in the queue, both Flush() and TaskAction() method will be waiting on an event to be set. Although the Flush() waits for 1ms only, but it is likely to happen that they are both waiting for a signal. And, when the queue is flushing all the pending messages, the only signal setter is the Run() method. `resetEvent.Set()` will only allow at most one thread to proceed who is waiting on it. Unfortunately, this may deviate from the intended behavior during flushing. That is, process all the pending messages and wait till they are all done. Would you consider adding another event that will notify the end of processing and then return from Flush() method?